client/fetch_state_pool.go

```diff
@@ -33,9 +33,21 @@ type fetchStatePoolImpl struct {
 }
 
 func newfetchStatePool(opts pool.ObjectPoolOptions) fetchStatePool {
```

Comment:
nit: This should maybe be `newFetchStatePool` for camel casing purposes (I know the resulting type is not capitalized, still though seems that should affect the camel casing convention perhaps?).

Reply:
for sure, definitely a typo.